lib/eslint.js

```diff
@@ -807,20 +811,18 @@ module.exports = (function() {
              */
             controller.traverse(ast, {
                 enter: function(node, parent) {
-
                     var comments = api.getComments(node);
 
                     emitCommentsEnter(comments.leading);
                     node.parent = parent;
-                    api.emit(node.type, node);
+                    codePathAnalyzer.enterNode(node);
```

Comment:
I'm not sure it's a good idea to have `codePathAnalyzer` emit the normal node events. Maybe have `enterNode()` return the code path and segment so eslint can emit them?